### PR TITLE
164139701 Add reading stats feature

### PR DESCRIPTION
**Pull request by youngestdj**
#### What does this PR do?
This PR adds a feature to allow a user see their reading stats
#### Description of task to be completed?
* Add endpoint that returns the total number of articles a user has read
#### How can this be manually tested?
* Create two user accounts
* Use the first account to create some articles
* Use the second account to get those articles by their slugs
* Send a get request to `GET /api/v1/user/readingstats` with a valid token belonging to the second user
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139701

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -222,4 +222,27 @@ export default class UserController {
...
+    try {
+      const viewCount = await user.getView();
+      return res.status(200).json({
+        success: true,
+        message: viewCount.length
+      });
+    } catch (error) {
+      return res.status(500).json({
+        success: false,
+        message: error.message
```
Comment (codinger41) [inline controllers/user.js]: Let's change this error message here to the right format.

File: routes/v1.js [hunk 2]
```diff
@@ -145,6 +145,7 @@ apiRoutes.get(
 
 apiRoutes.get(
   '/articles/:slug',
+  Auth.isLoggedIn,
```
Comment (codinger41) [inline routes/v1.js:158]: Let's do this in a way that if the user is logged in, you get the stats, but if the user isn't logged in, no stats. Or just anonymous readers.

Comment (youngestdj) [reply routes/v1.js:158]: That's exactly what is happening in that block. The middleware sends the user details in `req.user` if they are logged in and returns `next()` if they're not.

**Comment by codinger41** [comment]
Oh great then.

File: controllers/user.js (hunk 1 above)
Comment (youngestdj) [reply controllers/user.js]: Fixed, thanks for pointing it out

**Review by codinger41 — APPROVED** [review]

**Review by kleva-j — APPROVED** [review]

**Comment by kleva-j** [comment]
Resolve your conflicts.

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**